---
title: "Denoising and decontaminating v2"
output:
  html_document:
    df_print: paged
---
After running the Nextera_dada2 , we have to denoise the whole dataset. We will do this by using 4 different processes:


  * **Estimation of *Tag-jumping* or indices *cross-talk* **. We run multiple samples on each MiSeq run. These are identified by two sets of molecular barcodes. There is the potential of some sequences to be assigned to the wrong sample, which is a bummer. To estimate how many reads did this, on each MiSeq run we added some samples whose composition is known and extremely unlikely to be present in the enviromental samples studied. AS a result of this **Tag-jumping**, some of the positive control sequences might show in the environmental samples and viceversa. In our case, these positive controls are made of either Kangaroo or Ostrich (and Alligator). The process consists on, for each run, to model the compositon observed on the positive controls and substract it from the environmental samples from that run. The output will be a dataset with the same number of samples as before, but with fewer reads of certain sequences (ASVs)
  
  * **Discarding samples with extremely low number of reads**. Sometimes the number of reads sequenced from a particular replicate are really low, and hence the relative proportions of ASVs would be skewed. 
  
  * **Full clearance from Positive control influence**. THis process also takes advantage of the known composition of the positive controls. Each ASV found in the positive controls with a higher abundace in them than in the rest of the samples will be labelled as  **Positive** and removed from the environmental dataset. The output will be a dataset with the same number of samples as before but with fewer ASVs.
  
  * **Occupancy modelling** . Is the presence of a ASV a reflection of a biological reality or likely a PCR artifact? This may seem trivial in extreme cases (an ASV that only appears in one PCR replicate in the whole dataset) but how to discriminate between PCR artifacts from rare but real organisms? We use Occupancy modelling to determine if the pattern of presence of a ASV in a dataset reflects that. The output of this procedure will be a datasetwith the same number of samples as before but with fewer ASVs.
  
  * **Dissimilarity between PCR replicates**. The workflow that leads to the sequencing of a particular sample is subject to many stochatic processes, and it is not unlikely that the composition retrieved is very different for the original community. A way to ensure that this difference is minimal is through the separate analysis of each PCR replicate. We used that approach and modeled the dissimilarity between each PCr replicate and the group centroid. This way of modeling the dissimilarity allows us to discard those PCR replicate that won't fit the normal distribution of dissimilarities. The output of this procedure will be a dataset with the same number of **Hashes** as before but with fewer **samples**.
  
  
As with everything, we will start the process by loading the required packages and datasets.

# Load the dataset and metadata



```{r load libraries, include=FALSE}
 knitr::opts_chunk$set(warning = FALSE, message = F)

 library (tidyverse)
 library (vegan)
 #library (MASS)
 library (proxy)
 library (reshape2)
 library (seqinr)
 library (patchwork)
library(here)

```

We will load the ASV table and the metadata file. They are in the same folder so we use `list.files` to access them and a neat combination of `bind.rows` and `map(read_csv)`. 

```{r load datasets - we will be doing that for all runs}

metadata <- read_csv(here("Data", "eDNA.samples.with.lat.lon.csv"))

Run7 <- read_csv(here("Data/Pipeline_output/Run7/output_no_MaxEE/ASV_table.csv"))

Run8 <- read_csv(here("Data/Pipeline_output/Run8/output/ASV_table.csv"))

classified.hashes <- read_csv(here("Data/Pipeline_output/Run7/output_no_MaxEE/All.hash.classified.csv"))


all.hashes <- list(Run7 = read_csv(here("Data/Pipeline_output/Run7/output_no_MaxEE/hash_key.csv")),
                   Run8 = read_csv(here("Data/Pipeline_output/Run8/output/hash_key.csv")))

ASV.table <- bind_rows(list(Run7 = Run7, Run8 = Run8), .id = "Miseq_run")


Hash.key <- bind_rows(all.hashes) %>% distinct

final.metadata <- read_csv(here("Data", "Final_metadata.csv"))

# Write the Hash key from start

Hash.key %>% 
  write_csv(here("Output", "starting.hash.key.csv"))

```


## Data Cleanup - Don't act like you don't need this

Of the sequenced samples, let's check that all samples have the format Sample.rep. If they appear in the Run8 with the same name, let's change it

```{r data cleaning}
ASV.table %>% distinct(Sample_name, Miseq_run) %>% group_by(Sample_name) %>% tally(sort = T)

# There are 7 samples with the same name in both runs - update the 
# names in Run8

ASV.table %>% 
  mutate(Sample_name = case_when(Miseq_run == "Run8" & Sample_name == "1004-1" ~ "1004-3",
                                 Miseq_run == "Run8" & Sample_name == "489-2" ~ "489-3",
                                 Miseq_run == "Run8" & Sample_name == "81-2" ~ "81-4",
                                 Miseq_run == "Run8" & Sample_name == "848-2" ~ "848-4",
                                 Miseq_run == "Run8" & Sample_name == "985-1" ~ "985-4",
                                 Miseq_run == "Run8" & Sample_name == "998-2" ~ "998-4",
                                 Miseq_run == "Run8" & Sample_name == "999-1" ~ "999-3",
                                 TRUE                                         ~ Sample_name)) -> ASV.table # %>% # And check that now that does not happen
# distinct(Sample_name, Miseq_run) %>% group_by(Sample_name) %>% tally(sort = T) 
```
```{r}

ASV.table %>% 
  group_by(Miseq_run, Sample_name, Hash) %>% 
  summarise(nReads = sum(nReads)) %>% 
  ungroup -> ASV.table

ASV.table %>% 
  group_by(Hash) %>% 
  add_tally(wt = nReads) %>% 
  filter (n ==1)

ASV.table %>% 
  write_csv(here("Output", "starting.ASV_table.csv"))
```

The output of this process are a clean ASV table and a clean metadata file.

## Cleaning Process 1: Estimation of *Tag-jumping* or sample *cross-talk*

Before we modify our datasets on any way, we can calculate how many sequences that were only supposed to be in the positives control appeared in the environmental samples, and how many did the opposite. First we divide the dataset into positive control and environmental samples. Also create an ordered list of the Hashes present in the positive controls, for ease of plotting

```{r split into two}

ASV.table %>%  mutate(source = case_when(str_detect(Sample_name, "Tilapia")    ~   "Positives",
                                         TRUE                                                ~   "Samples")) -> ASV.table

ASV.table %>% 
  filter (source == "Positives") %>% 
  group_by(Hash) %>% 
  summarise(tot = sum(nReads)) %>% 
  arrange(desc(tot)) %>% 
  pull(Hash) -> good.order


```

Now let's create a jumping vector. What proportion of the reads found in the positives control come from elsewhere, and what proportion of the reads in the samples come from the positives control.

### Step 1: Nest the dataset and split it in positives and samples

To streamline the process and make it easier to execute it similarly but independently on each Miseq run, we nest the dataset by run. 
So Step1 is create a nested table so we can run this analysis on each run independently. 


```{r nesting the dataset}

ASV.table %>% 
  group_by(Miseq_run, source) %>% 
  nest() %>% 
  pivot_wider(names_from = source, values_from =  data) -> ASV.nested 
```

That wasn't too complicated. Let's start a summary function that keeps track of our cleaning process

```{r summary.file}

how.many <- function(ASVtable, round){
  ASVtable %>% ungroup() %>% 
    summarise(nsamples = n_distinct(Sample_name),
              nHashes = n_distinct(Hash),
              nReads = sum(nReads), 
              Stage = paste0("Step_", round)) %>% 
    gather(starts_with("n"), value = "number", key = "Stat")
}

ASV.nested %>% 
  transmute(Summary = map(Samples, ~ how.many(ASVtable = .,round = 0)))  -> ASV.summary

```

### Step 2: Model the composition of the positive controls of each run 


We create a vector of the composition of each positive control and substract it from the environmental samples from their runs



```{r jumping vector}


ASV.nested %>% 
  mutate (contam.tibble = map(Positives, 
                              function(.x){
                                .x %>%
                                  group_by(Sample_name) %>%
                                  mutate (TotalReadsperSample = sum(nReads)) %>%
                                  mutate (proportion = nReads/TotalReadsperSample) %>%
                                  group_by (Hash) %>%
                                  summarise (vector_contamination = max (proportion))
                                }) ) -> ASV.nested

ASV.nested %>% 
  group_by(Miseq_run) %>% 
  select(contam.tibble) %>% 
  unnest(cols = contam.tibble) # Check how it looks like



```


### Step 3: Substract the composition of the positive controls from the environment samples

The idea behind this procedure is that we know, for each run, how many reads from each Hash appeared in the positive controls. These come from 2 processes: sequences we know should appear in the positive controls, and sequences that have *jumped* from the environment to the positive controls. With this procedure, we substract from every environmental sample the proportion of reads that jumped from elsewhere.

```{r cleaning step 1}
ASV.nested %>% 
  mutate(cleaned.tibble = map2(Samples, contam.tibble, function(.x,.y){ 
    .x %>%
      group_by (Sample_name) %>%
      mutate (TotalReadsperSample = sum (nReads)) %>%
      left_join(.y, by = "Hash") %>%
      mutate (Updated_nReads = ifelse (!is.na(vector_contamination),  nReads - (ceiling(vector_contamination*TotalReadsperSample)), nReads)) %>%
      filter (Updated_nReads > 0) %>%
      ungroup() %>% 
      dplyr::select (Sample_name, Hash, nReads = Updated_nReads)
      
    
  })) -> ASV.nested

ASV.nested %>% 
  group_by(Miseq_run) %>% 
  select(cleaned.tibble) %>% 
  unnest(cleaned.tibble) #Check how they look


```
Add this step to the summary table we were creating

```{r summary.file.2}
ASV.nested %>% 
  transmute( Summary.1 = map(cleaned.tibble, ~ how.many(ASVtable = .,round = "1.Jump"))) %>% 
  left_join(ASV.summary) %>% 
  mutate(Summary   = map2(Summary, Summary.1, bind_rows)) %>%
  dplyr::select(-Summary.1) -> ASV.summary 
```

## Cleaning Process 2: **Discarding PCR replicates with low number of reads**

We will fit the number of reads assigned to each sample to a normal distribution and discard those samples with a probability of 95% of not fitting in that distribution. The output would be a dataset with less samples and potentially less number of unique Hashes.

```{r fitting nReads per sample}

ASV.nested %>% 
  select(Miseq_run,cleaned.tibble) %>% 
  unnest(cleaned.tibble) %>% 
  group_by(Sample_name) %>%
  summarise(tot = sum(nReads)) -> all.reps

# Visualize

all.reps %>%  
  pull(tot) -> reads.per.sample

names(reads.per.sample) <- all.reps %>% pull(Sample_name)  

normparams.reads <- MASS::fitdistr(reads.per.sample, "normal")$estimate



all.reps %>%  
  mutate(prob = pnorm(tot, normparams.reads[1], normparams.reads[2])) -> all.reps

#  probs <- pnorm(all_pairwise_distances, normparams[1], normparams[2])

outliers <- 
  all.reps %>% 
  filter(prob < 0.075 & tot < normparams.reads[1])

ASV.nested %>% 
  mutate(Step.1.low.reads = map (cleaned.tibble, ~ filter(.,!Sample_name %in% outliers$Sample_name) %>% ungroup)) -> ASV.nested

ASV.nested %>% 
  transmute( Summary.1 = map(Step.1.low.reads, ~ how.many(ASVtable = .,round = "2.Low.nReads"))) %>% 
  left_join(ASV.summary) %>% 
  mutate(Summary   = map2(Summary, Summary.1, bind_rows)) %>%
  dplyr::select(-Summary.1) -> ASV.summary 

```




## Cleaning Process 3: **Full clearance from Positive control influence**

Removing the Hashes that belong to the positive controls. First, for each Hash that appeared in the positive controls, determine whether a sequence is a true positive or a true environment. For each Hash, we will calculate, maximum, mean and total number of reads in both positive and samples, and then we will use the following decission tree:

  * If all three statistics are higher in one of the groups, we will label it either of Environmental or Positive control influence.
  
  * If there are conflicting results, we will use the Hashes. to see if they belong to either the maximum abundance of a Hash is in a positive, then it is a positive, otherwise is a real sequence from the environment.


Now, for each Hash in each set of positives controls, calculate the proportion of reads that were missasigned - they appeared somewhere they were not expected.
We will divide that process in two: first . A second step would be to create a column named proportion switched, which states the proportion of reads from one Hash that jumped from the environment to a positive control or viceversa. The idea is that any presence below a threshold can be arguably belong to tag jumping.

```{r real or positive}


ASV.table %>% 
  filter (Hash %in% good.order) %>%
  group_by(Sample_name) %>% 
  mutate(tot.reads = sum(nReads)) %>% 
  group_by(Hash,Sample_name) %>% 
  mutate(prop = nReads/tot.reads) %>% 
  group_by(Hash, source) %>% 
  summarise (max.  = max(prop),
             mean. = mean(prop),
             tot.  = sum(nReads)) %>% 
  gather(contains("."), value = "number", key = "Stat") %>%
  spread(key = "source", value = "number", fill = 0) %>% 
  group_by(Hash, Stat) %>%
  mutate(origin = case_when(Positives > Samples ~ "Positive.control",
                            TRUE                ~ "Environment")) %>% 
  group_by (Hash) %>%
  mutate(tot = n_distinct(origin)) -> Hash.fate.step2

Hash.fate.step2 %>% 
  filter(tot == 1) %>% 
  group_by(Hash) %>% 
  summarise(origin = unique(origin)) %>% 
  filter(origin == "Positive.control") -> Hashes.to.remove.step2

ASV.table %>% 
  group_by(source, Hash) %>% 
  summarise(ocurrences =n()) %>% 
  spread(key = source, value = ocurrences, fill = 0) %>% 
  #left_join(Hashes.to.remove.step2) %>% 
  #mutate(origin = case_when(is.na(origin) ~ "Kept",
   #                         TRUE          ~ "Discarded")) %>% 
  mutate(second.origin = case_when(Positives >= Samples ~ "Discarded",
                                   TRUE                 ~ "Kept")) %>% 
  filter(second.origin == "Discarded") %>% 
  full_join(Hashes.to.remove.step2) -> Hashes.to.remove.step2

# Hashes.to.remove.step2 %>% 
#   bind_rows(tibble(Hash = c("01204d587a3c3752f426f2f6d813c0ff2b80ec8b",
#                             "acebcd5c491bb273f3e4d615cafad649"))) -> Hashes.to.remove.step2
#  

```
IN order to train DADA2 to better distinguish when positive control sequences have arrived in the environment, we will keep the sequences in a csv file


```{r ASVs from positives}

Hashes.to.remove.step2 %>% 
  left_join(Hash.key) %>% 
  select(Hash, Sequence) %>% 
  write_csv("../Output/Hashes.to.remove.csv")

```

### Remove the positive control hashes from the composition of the ASVs

```{r cleaning.Step2}

ASV.nested %>% 
  mutate(Step2.tibble = map (Step.1.low.reads, ~ filter(.,!Hash %in% Hashes.to.remove.step2$Hash) %>% ungroup)) -> ASV.nested

saveRDS(ASV.nested, file = "../Output/Cleaning.before.Occ.model")

ASV.nested <- readRDS(file ="../Output/Cleaning.before.Occ.model")

ASV.nested %>% 
  transmute( Summary.1 = map(Step2.tibble, ~ how.many(ASVtable = .,round = "3.Positives"))) %>% 
  left_join(ASV.summary) %>% 
  mutate(Summary   = map2(Summary, Summary.1, bind_rows)) %>%
  dplyr::select(-Summary.1) -> ASV.summary 

ASV.summary %>% 
  unnest()


```

<!-- ## Cleaning Process 4: **Occupancy modelling** -->

<!-- What is the probabilty of a true positive presence of a Hash in a Miseq Run. We will use eDNA occupancy modeling to asses whether a hash is a rare variant that spilled out or a true presence. -->

<!-- The process requires to load extra packages, create some model file, and group the hashes by Run, and biological replicate, summarising the data in a presence absence format. -->

<!-- The occupancy model itself was performed in the Rmarkdown file `Rjags.tunning.Rmd`, so here we will upload the csv file that contains all probability of occurences of all hashes per site. Each Hash-Site combination produces a matrix of presence abascences that feeds the model - for some cases it is a 30x3 matrix, for others it is a 39x3. We summarised the number of occurences in each case and run models for each unique case (to save computing time). Each unique model was run 10 times to filter out cases in which the model converge into a local maxima. -->

<!-- So we will import the object `Occ.fate.csv` and reduce the dataset to those Hashes with an occ > 0.8 -->

<!-- ```{r importing Occ results} -->

<!-- occ.results <- read_csv("../Input/Occ.fate.2020.csv") -->

<!-- occ.results %>%  -->
<!--   ggplot(aes(x = max)) + -->
<!--   geom_histogram(fill = "blue") + -->
<!--   labs(x = "", -->
<!--        y = "Number of Hashes")  -> left.plot -->


<!-- occ.results %>%  -->
<!--   left_join(ASV.nested %>%  -->
<!--               select(Step2.tibble) %>%  -->
<!--               unnest(cols = Step2.tibble) %>%  -->
<!--               group_by(Hash) %>%  -->
<!--               summarise (tot = sum(nReads))) %>%  -->
<!--   ggplot(aes(x = cut_interval(max, n = 20))) + -->
<!--   geom_col(aes(y = tot), fill = "red", position = "stack") + -->

<!--   labs(x = "", -->
<!--        y = "Number of Reads")   -> right.plot -->


<!-- left.plot + right.plot  -->

<!-- ``` -->

<!-- So we will throw away most of the Hashes, but will keep most of the reads - we are getting into something here -->

<!-- ```{r actual filtering} -->
<!--  occ.results %>%  -->
<!--   filter(max > 0.8) %>%  -->
<!--   pull (Hash) -> to.keep -->

<!-- ASV.nested %>%  -->
<!--   mutate(Step3.tibble = map (Step2.tibble, ~ filter(.,Hash %in% to.keep))) -> ASV.nested -->

<!-- ASV.nested %>%  -->
<!--   transmute(Summary.1 = map(Step3.tibble, ~ how.many(ASVtable = .,round = "4.Occupancy"))) %>%  -->
<!--   left_join(ASV.summary) %>%  -->
<!--   mutate(Summary   = map2(Summary, Summary.1, bind_rows)) %>% -->
<!--   dplyr::select(-Summary.1) -> ASV.summary  -->


<!-- ``` -->


## Cleaning Process 5: **Dissimilarity between PCR replicates**

So, a second way of cleaning the dataset is to remove samples for which the dissimilarity between PCR replicates exceeds the normal distribution of dissimilarities.
Sometimes the preparation of a PCR replicate goes wrong for a number of reasons - that leads to a particular PCR replicate to be substantially different to the other 2. In that case, we will remove the PCR replicate that has higher dissimilarity with the other two.

The process starts by adding the biological information to the ASV table, then diving the dataset by their biological replicate. This will also remove any sample that is not included in the metadata, eg coming from a different project.

```{r dissimilarity between PCR replicates}

ASV.nested %>% 
  select(Miseq_run, Step2.tibble) %>% 
  unnest(Step2.tibble) %>%
  separate(Sample_name, into = c("original_sample", "rep"), remove = F) -> cleaned.tibble
```


```{r quick check}
# do all samples have a name
cleaned.tibble %>% 
  filter (Sample_name == "")
# do all of them have an original sample
cleaned.tibble %>% 
  filter(original_sample == "")
# do all of them have a Hash
cleaned.tibble %>% 
  filter(is.na(Hash))
# How many samples, how many Hashes
cleaned.tibble %>%
  ungroup %>% 
  summarise(n_distinct(Sample_name), # 225
            n_distinct(Hash))   # 2462

# Let's check the levels of replication

cleaned.tibble %>% 
  group_by(original_sample) %>% 
  summarise(nrep = n_distinct(Sample_name)) %>% #115 uniquesam[les]
  filter (nrep == 2) # 77
  # filter (nrep == 1) # 25 
```
Ok, so there are 13 samples for which we only have 2 PCR replicates1.   We will get rid of those with only 1, as we can't estimate the PCR bias there. THis is 

```{r remove single replicates}
discard.1 <- cleaned.tibble %>% 
  group_by(original_sample) %>% 
  mutate(nrep = n_distinct(Sample_name)) %>% 
  #filter (nrep == 2) # 25
  filter (nrep == 1) %>% 
  distinct(Sample_name) %>% pull(Sample_name)

cleaned.tibble %>% 
  filter(!Sample_name %in% discard.1) -> cleaned.tibble_ready
```

Anyway, let's have a visual representation of the dissimilarities between PCR replicates, biological replicates and everything else.

```{r lets do the PCR replication}
cleaned.tibble_ready %>%
  group_by (Sample_name) %>%
  mutate (Tot = sum(nReads),
          Row.sums = nReads / Tot) %>% 
  group_by (Hash) %>%
  mutate (Colmax = max (Row.sums),
          Normalized.reads = Row.sums / Colmax) -> cleaned.tibble_ready
tibble_to_matrix <- function (tb) {
  
  tb %>% 
    group_by(Sample_name, Hash) %>% 
    summarise(nReads = sum(Normalized.reads)) %>% 
    spread ( key = "Hash", value = "nReads", fill = 0) -> matrix_1
    samples <- pull (matrix_1, Sample_name)
    matrix_1 %>% 
      ungroup() %>% 
    dplyr::select ( - Sample_name) -> matrix_1
    data.matrix(matrix_1) -> matrix_1
    dimnames(matrix_1)[[1]] <- samples
    vegdist(matrix_1) -> matrix_1
}

tibble_to_matrix (cleaned.tibble_ready) -> all.distances.full

# Do all samples have a name?
summary(is.na(names(all.distances.full))) # Yes they do
```

Let's make the pairwise distances a long table
```{r}

as_tibble(subset(melt(as.matrix(all.distances.full)))) -> all.distances.melted

# Any mjor screw ups
summary(is.na(all.distances.melted$value))

# Now, create a three variables for all distances, they could be PCR replicates, BIOL replicates, or from the same site

all.distances.melted %>%
 
  separate (Var1, into = c("eDNA.sample","rep"), remove = FALSE, convert = T) %>%
  left_join(metadata) %>% #filter (is.na(Transect)) all joined correctly
  separate (Var2, into =c("eDNA.sample2","rep2"), remove = FALSE, convert = T) %>%
  left_join(suffix = c("1","2"),metadata, by = c("eDNA.sample2"= "eDNA.sample")) %>%
  unite(Transect1, position1, depth1, col = "BIOL1", remove = F) %>% 
  unite(Transect2, position2, depth2, col = "BIOL2", remove = F) %>% 
  unite(Transect1, position1,  col = "Stop1", remove = F) %>% 
  unite(Transect2, position2, col = "Stop2", remove = F) %>%
  mutate ( 
    
    Distance.type = case_when( eDNA.sample == eDNA.sample2 ~ "PCR.replicates",
                                      BIOL1 == BIOL2 ~ "Biol.replicates",
                                      Stop1 == Stop2 ~ "Same Stop",
                                      TRUE ~ "Different Site"
                                     )) %>%
  dplyr::select(Sample1 = Var1, Sample2 = Var2 , value , Distance.type) %>%
  filter (Sample1 != Sample2) -> all.distances.to.plot

# Checking all went well

sapply(all.distances.to.plot, function(x) summary(is.na(x))) # good boi

all.distances.to.plot$Distance.type <- all.distances.to.plot$Distance.type  %>% fct_relevel( "PCR.replicates", "Biol.replicates", "Same Stop")

  ggplot (all.distances.to.plot ) +
  geom_histogram (aes (fill = Distance.type, x = value, after_stat(ndensity)), position = "dodge",  alpha = 0.9, bins = 50) +
  facet_wrap( ~ Distance.type) +
  labs (x = "Pairwise dissimilarity", y = "density" ,
        Distance.type = "Distance") +
    guides (fill = "none")
ggsave("visual.anova.png", dpi = "retina")
```

So our the distribution of dissimilarities is as we expected : lowest in technical replicates, then biological replicates and higher across our study system. Now let's see if there are any technical replicates that should be discarded due to their higher dissimilarity. We will calculate the distance from each PCR replicate to their group centroid, fit those distances to a normal distribution and discard values that are too high

```{r}
# Instead of chosing based on the pw distances, we can do a similar thing using the distance to centroid

# Find out which samples have only two pcr replicates
cleaned.tibble_ready %>% dplyr::select(-Miseq_run) %>% mutate(eDNA.sample = as.numeric(original_sample)) %>% 
  left_join(metadata) %>% 
  unite(Transect, position, depth, col = "biol") %>% 
  group_by(biol) %>% nest() -> nested.cleaning

nested.cleaning %>% 
  mutate(matrix = map(data, tibble_to_matrix)) -> nested.cleaning
nested.cleaning %>% mutate(ncomparisons = map_dbl(matrix, length)) -> nested.cleaning
 
  
dist_to_centroid <- function (x,y) {
  biol <- rep(y, length(names(x)))
  
  if (length(biol) == 1) {
    output = rep(x[1]/2,2)
    names(output) <- attr(x, "Labels")
  }else{ 
    
  dispersion <- betadisper(x, group = biol)
  output = dispersion$distances
  }
  output
    }


nested.cleaning <- nested.cleaning %>% mutate (distances = map2(matrix, biol, dist_to_centroid))

unlist (nested.cleaning$distances) -> all_distances

hist(all_distances)
```

```{r}
#normparams <- fitdistr(all_pairwise_distances, "normal")$estimate
normparams <- MASS::fitdistr(all_distances, "normal")$estimate                                      
#  probs <- pnorm(all_pairwise_distances, normparams[1], normparams[2])
probs <- pnorm(all_distances, normparams[1], normparams[2])
outliers <- which(probs>0.95)

discard <-names (all_distances[outliers])


to_write_discarded <- as.data.frame(all_distances[outliers]) %>% rownames_to_column("sample") %>% dplyr::select(sample, 
                                                                                                     distance_to_centroid = 2)
# to_write_discarded <- to_write_discarded %>% bind_rows(tibble(sample = discard.1,
#                                                               distance_to_centroid = NA))
write_csv(to_write_discarded ,"../Output/discared_samples.csv")

# Who passes this filter

all_distances %>%
  as_tibble() %>% 
  mutate(sample = names(all_distances)) %>% 
  anti_join(to_write_discarded) %>% 
  separate(sample, into = "eDNA.sample", convert = T) %>% 
  group_by(eDNA.sample) %>% 
  summarise(cases = n()) %>% 
  left_join(final.metadata)  -> Coverage.dataset 

Coverage.dataset %>% 
  group_by(Openness, position, Transect) %>% 
  summarise(cases = sum(cases)) %>% 
  ggplot()+
  geom_raster(aes(x= Openness, y = position, fill = cases))+
  geom_text(aes(x= Openness, y = position, label = cases)) +
  facet_wrap(~Transect, ncol = 1, scales = "free_y")
  
  
  
  
```

Finally, let's remove these samples from the dataset

```{r actual cleaning}

ASV.nested %>% 
  mutate(Step3.tibble = map (Step2.tibble,  ~ anti_join(., to_write_discarded, by = c("Sample_name" = "sample")))) -> ASV.nested

ASV.nested %>% 
  transmute(Summary.1 = map(Step3.tibble, ~ how.many(ASVtable = .,round = "4.PCR.dissimilarity"))) %>% 
  left_join(ASV.summary) %>% 
  mutate(Summary   = map2(Summary, Summary.1, bind_rows)) %>%
  dplyr::select(-Summary.1) -> ASV.summary 
```


## Exporting the output

We will export the final cleaned table with four columns (Miseq_run, sample, Hash, nReads)

```{r}

ASV.nested %>% 
  select(Miseq_run, Step3.tibble) %>% 
  unnest(Step3.tibble) %>% 
  mutate(nReads = as.integer(nReads)) %>% 
  write_csv("../Output/ASV_table_all_together.csv")

ASV.nested %>% 
  select(Step3.tibble) %>% 
  ungroup() %>% 
  unnest(Step3.tibble) %>% 
  distinct(Hash) %>% 
  left_join(Hash.key) %>% 
  write_csv("../Output/Hash_Key_all_together.csv")



input <- read_csv("../Output/Hash_Key_all_together.csv")
output <- "../Output/Hash_Key_all_together.fasta"

write.fasta (sequences = as.list(input$Sequence),
             names = as.list(input$Hash),
             file.out = output)


```

## Summary of the cleanup process

```{r last graph}

ASV.summary %>% 
  unnest() %>% 
  ggplot(aes(x=Stage, y=number, fill = Stat))+
    geom_line(aes(group = Miseq_run, color = Miseq_run))+
  facet_grid(Stat~., scales = "free")+
  theme(axis.text.x = element_text(angle = 45, hjust =1))#,
                                 


```

# Coverage Table with nReads

```{r}

ASV.table <- read_csv ("../Output/ASV_table_all_together.csv")
ASV.table %>% 
  separate(Sample_name, into = "eDNA.sample", sep = "-", convert = T) %>% 
  left_join(final.metadata) %>% 
  group_by(Openness, position, Transect) %>% 
  summarise(nReads = sum(nReads),
            hashes = n_distinct(Hash)) -> midstep
    

# As a table
midstep %>% 
  pivot_wider(names_from = Date, values_from = data, id_cols = c(Area, Site),values_fill = list(data = "") ) %>% 
  slice(match(Site, order.Sites)) %>% 
  kableExtra::kable() %>% 
  kableExtra::kable_styling() %>% 
  kableExtra::collapse_rows(1, valign = "top") 

# As a graph
midstep %>% 
  right_join(Coverage.dataset) %>% 
  group_by(event) %>% 
  mutate(mean = mean(nReads),
         sum = sum(nReads)) %>% 
  mutate(Site = factor(Site, levels = order.Sites)) %>% 
  ggplot(aes(x = Date, 
             y = nReads)) +
  geom_boxplot(aes(group=Date),outlier.alpha = 0.5) +
  geom_point(aes(y = sum,
                 size = cases,
                 color = Area)) +
  scale_y_continuous(labels = scales::comma) +
  scale_x_date(labels = function(x){paste(lubridate::month(x, label = T),
                      lubridate::year(x),
                      sep= "'")}) +
  labs(y = "#reads", x = "") +
  
  facet_wrap(~Site , nrow = 5,dir = "v", drop = T) +
  theme_minimal() +
  theme(axis.title.y = element_text(angle = 0,vjust = 0.5),
        legend.position = c(0.6, 0.2)) 

```

