---
title: "By group"
author: "Ramon Gallego"
date: "7/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Plan

Let's get an idea of the ASVs by family
```{r cars}
library (tidyverse)
library (here)
library (ape)
library (insect)

```

## Get data

Load the hash_key, the classified hashes & ASV

```{r pressure, echo=FALSE}

ASV.table <- read_csv(here ("Output","ASV_table_all_together.csv"))

Classification <- read_csv(here("Data", "Final_classification.csv"))

Sequences <- readFASTA(here("Output","Hash_Key_all_together.fasta"))

Seqs <- read_csv(here("Output","Hash_Key_all_together.csv"))

```

## Does Swarm solve the issue?

```{r}
ASV.table %>% 
  group_by(Hash) %>% 
  summarise(nReads = sum(nReads)) -> size

size %>% 
  mutate(label = paste0(Hash, ";size=", nReads)) -> size

left_join(Seqs, size) -> for.swarm

seqinr::write.fasta(sequences = as.list (for.swarm$Sequence),
                    names = as.list(for.swarm$label),
                    file.out = here("Output", "swarm.input.fasta"))

```

Run Swarm. Do it directly on the terminal bc it does not find it from RStudio

<!-- ```{bash} -->

<!-- bash /Users/ramon.gallegosimon/Projects/banzai/scripts/OTU_clustering/cluster_swarm.sh \ -->
<!--  /Users/ramon.gallegosimon/Projects/eDNA-Hake/Output/swarm.input.fasta -->

<!-- ``` -->


Load the conversion table and the classification summary

```{r}
Swarm.conversion <- read_csv(here("Output", "OTUs_swarm","dups_to_otus.csv")) %>% 
  separate(Query, into = "Hash")
  
n_distinct(Swarm.conversion$Match)

```

```{r combine}

Combine.clas <- left_join(Classification, Swarm.conversion) %>% 
  separate(Match, into = "New.Hash") %>% 
  filter (Hash == New.Hash) 

n_distinct(Swarm.conversion$Match) # We had 443 Swarm OTUs
n_distinct(Combine.clas$New.Hash)  # We found taxonomy for 368 of them
Combine.clas %>% 
  group_by(New.Hash) %>% 
  add_tally() %>% 
  filter (n > 1) # There is one New.Hash with two confronting identifications

Combine.clas %>% 
  group_by(New.Hash) %>% 
  slice(1) -> Combine.clas #%>% filter (New.Hash == "e5b505705fe64fe02348aa9612ad44e82b0d800e")
  
  
  
Swarm.conversion %>% 
  separate(Match, into = "New.Hash") %>% 
  left_join(Classification, by = c("New.Hash" = "Hash")) -> new.Classification

new.Classification %>% # Now keep only one match from each New Hash
  # select(-New.Hash) %>% 
  group_by(Hash) %>% 
  slice(1)-> new.Classification #filter (New.Hash == "e5b505705fe64fe02348aa9612ad44e82b0d800e")

new.Classification %>%   
select(-New.Hash) %>% 
  write_csv(here("Output", "Hashes.classified.swarm.csv"))


Combine.clas %>%  group_by(origin, is.na(species)) %>% 
  tally(sort = T)
Classification %>% group_by(origin, is.na(species)) %>% 
  tally(sort = T)


  
```

## Measuring success.

For each ASV, record prevalence and nReads. Then see the effect of swarm in getting more or less species-level ids

```{r}
ASV.table %>% 
  group_by(Hash) %>% 
  summarise (n = n_distinct(Sample_name),
             nReads = sum(nReads)) -> summary

Classification.success <- function (ASVsummary, key){
  
  left_join(ASVsummary, key) %>% 
    group_by(`Species-level` = !is.na(species)) %>% 
    summarise (nReads = sum(nReads),
               nASVs = n(), 
               nspp = n_distinct(species)
    )
}

Classification.success(summary,Classification )

Classification.success(summary, new.Classification)

```

Using swarm:

 - Reduces the number of reads and ASVs not classified to species level
 - Reduces the number of distinct species seen
 
```{r}
left_join(summary, Classification) %>% 
  distinct(species) %>% 
  anti_join(
    
    left_join(summary, new.Classification)
  )
```
 
Particularly within Seabastes, where the interspecies distances are really small

# Hakes

```{r}
left_join(ASV.table, new.Classification) %>% 
  filter (str_detect(family, "Merlucc")) %>% 
  group_by(species ) %>% 
  summarise(n_distinct(Hash),
            sum(nReads),
            n_distinct(Sample_name))
```

```{r}
left_join(ASV.table, Classification) %>% 
  filter (str_detect(family, "Merlucc")) %>% 
  group_by(species ) %>% 
  summarise(n_distinct(Hash),
            sum(nReads),
            n_distinct(Sample_name))
```

So, of the 4 ASVs identified only to genus, 1 went to the M productus cluster, the other 3 to the productus-angustimanus

## Group by family, generate heatmaps

```{r}
dna.dist.possible <- possibly(dist.dna, otherwise = NA)

dist.dna
names(Sequences) == Seqs$Hash
 
new.Classification %>% 
  group_by(Hash) %>% 
  add_tally() %>% 
  filter (n >1)

left_join(Seqs, new.Classification) %>% 
  pull(family) -> fams
str(fams)

split(Sequences, fams) ->by.fam

tibble(by.fam) %>% mutate(family = names(by.fam)) -> by.fam.tibble

by.fam.tibble %>% 
  mutate(nseqs = map_dbl(by.fam, length)) %>% 
  slice(2) %>%
  group_by(family) %>% 
  mutate(lengths =  map(by.fam,function(.x) map_dbl(.x, length)),
          lengths = map(lengths,~ bind_cols(.x) %>% mutate(seq = names(.x)) )
         ) %>% 
  pull(lengths)
  map(dna.dist.possible, model = "N", pairwise.deletion = T, as.matrix = T)

dna.dist.possible <- possibly(dist.dna, otherwise = NA)
map(by.fam, dna.dist.possible, model = "N", pairwise.deletion = T, as.matrix = T) %>% 
  map(reshape2::melt, value.name = "dissimilarities") ->list.of.dfs
  
list.of.dfs %>% 
  compact() %>% 
  # map(~.x %>% mutate (dissimilarities = case_when(is.character(dissimilarities)~ "200", 
  #                               TRUE                         ~ dissimilarities))) %>% 
  bind_rows(.id = "family") %>% 
  filter (!is.na(Var1)) %>% 
  ggplot(aes(x = Var1, y = Var2, fill = dissimilarities>1)) +
  geom_raster() +
  scale_fill_viridis_d()+
  theme(axis.text = element_blank()) +
  facet_wrap(~family, scales = "free")
  
list.of.dfs %>% 
  compact() %>% 
  # map(~.x %>% mutate (dissimilarities = case_when(is.character(dissimilarities)~ "200", 
  #                               TRUE                         ~ dissimilarities))) %>% 
  bind_rows(.id = "family") %>% 
  filter (!is.na(Var1)) %>% 
  ggplot(aes(x = dissimilarities)) +
  geom_histogram()
```

# Export the sequences

```{r}
new.Classification %>% 
  inner_join(Seqs) %>% 
  filter(!is.na(class)) %>% 
  filter (!is.na(family)) %>% 
  # filter (is.na(genus))
  group_by(family) %>% 
  nest() %>%
  mutate(
  output = walk2(family, data, function(.x,.y){
    .y %>% 
      unite(Hash, New.Hash, sep = ";NEW = ", col = "name") %>% 
      unite(name, species, sep = ";organism=", col = "name") -> temp
      
    seqinr::write.fasta(sequences = as.list(temp$Sequence),
                        names     = as.list(temp$name),
                        file.out  = here("Data","Fasta_families", paste0(.x, ".fasta")))
  })
  )
```

