% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={STAN for Thermocline},
  pdfauthor={Ramon Gallego},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{STAN for Thermocline}
\author{Ramon Gallego}
\date{8/26/2021}

\begin{document}
\maketitle

\hypertarget{making-a-stan-model-for-hypothesis-testing-of-metabarcoding-data}{%
\subsection{Making a STAN model for hypothesis testing of metabarcoding
data}\label{making-a-stan-model-for-hypothesis-testing-of-metabarcoding-data}}

We have one problem with the application of metabarcoding data for
hypothesis testing - we can't really be sure if the weight we are
putting on number of reads, sequencing depth data transformation and
algorithm to estimate sample dissimilarity are the ones driving the
change and therefore could mask the real biological effect. I include
below the libraries I used

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(here)}
\FunctionTok{library}\NormalTok{ (tidyverse)}
\FunctionTok{library}\NormalTok{(rstan)}
\end{Highlighting}
\end{Shaded}

\hypertarget{the-design}{%
\subsection{The design}\label{the-design}}

We have sampled eDNA from locations in the Pacific Coast, at different
depths. At each location and depth, we collected up to two Niskin
Bottles, and each of them was amplified and sequenced twice (some of
them more, some less).

The idea is to get to a posterior probability of the proportion of DNA
from each taxa (\(*i*\)) at each lat,lon,depth point (\(*j*\)), taking
into account the number of PCR cycles, the primer efficiency and the
stochasticity of the PCR process.

The Amount of DNA in the water from taxa \(*i*\) \(D_i\) is unknown but
we can express it as function of the number of Amplicons detected
\(A_i\) and the process that generates those: Amplification at the
efficiency (\(\alpha\)) for a number of PCR cycles. We don't see all the
amplicons we generate, only a proportion of them end up in the
sequencing machine. \(\eta\)

\begin{equation} 
  f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
  (\#eq:binom)
\end{equation}

\begin{equation}

 A_{ij} = C_{ij}(1+\alpha)^{N_{PCR}} \times  \eta 
 (\#eq:label)
\end{equation}

And in log space

\begin{equation}
log(A_{ij}) = log (C_{ij}) + N_{PCR} \times log(1+\alpha) + log (\eta)
\end{equation}

But we can express the original Counts as the proportion of DNA copies
from species \(*i*\) in the sample so

\begin{equation}
 B_{ij} = \frac{C_{ij}}{\sum_{i}C_{ij}} 
\end{equation}

\begin{equation}
log(B_{ij}) = log(C_{ij}) - log(\sum_i C_{ij})
\end{equation}

And the last term is the same for all values from sample \(*j*\).

So the equation that brings the

Now we change \(A_{ij}\) - which assumes a deterministic process from a
proportion of copies of DNA and number of PCR cycles, and thus make it
\(\lambda_{ijk}\) where \(*k*\) is the particular PCR reaction from that
\(*j*\) and the Observed values come from a negative binomial
distribution

\begin{equation}
log(\lambda_{ijkl}) = log(B_{ijkl})  + N_{PCR} \times log(1+\alpha) + log (\eta_{jkl}) \\ Y_{ijkl} \sim {\sf NegativeBinomial}(\lambda_{ijkl}, \tau)

\end{equation}

And the variance is \begin{equation}
\lambda_{ijkl} + \frac{\lambda^2_{ijkl}}{\tau}
\end{equation}

As it is, we are assuming that PCRs from two Niskins from the same
station reflect the same, (aka biological replicates are perfect). But
we can bring the concentration of DNA at site \(*j*\) (a combination of
lat, lon and depth) as:

\begin{equation}
log(\lambda_{ijl}) = \gamma_{il} + \delta_{ijl} \\ \delta_{ijl} âˆ¼ Normal(0,\sigma^2)
\end{equation}

\hypertarget{get-the-data-prepare-the-data}{%
\subsubsection{Get the data, prepare the
data}\label{get-the-data-prepare-the-data}}

You remember from previous adventures how picky STAN is with regards to
the structure of the data. starting with the tibbles with the metadata
and the counts, and we'll move from there

\end{document}
